tests/by-util/test_stat.rs

```diff
@@ -352,3 +352,80 @@ fn test_without_argument() {
         .fails()
         .stderr_contains("missing operand\nTry 'stat --help' for more information.");
 }
+
+#[test]
+fn test_quoting_style_locale() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+    at.touch("'");
+    ts.ucmd()
+        .env("QUOTING_STYLE", "locale")
+        .args(&["-c", "%N", "'"])
+        .run()
+        .no_stderr()
+        .stdout_is("'\\''\n")
+        .succeeded();
+
+    ts.ucmd()
+        .args(&["-c", "%N", "'"])
+        .run()
+        .no_stderr()
+        .stdout_is("\"'\"\n")
+        .succeeded();
+}
+
+#[test]
+fn test_printf_octal_1() {
+    let ts = TestScenario::new(util_name!());
+    let expected_stdout = vec![0x0A, 0xFF]; // Newline + byte 255
+    ts.ucmd()
+        .args(&["--printf=\\012\\377", "."])
+        .succeeds()
+        .stdout_is_bytes(expected_stdout);
+}
+
+#[test]
+fn test_printf_octal_2() {
+    let ts = TestScenario::new(util_name!());
+    let expected_stdout = vec![b'.', 0x0A, b'a', 0xFF, b'b'];
+    ts.ucmd()
+        .args(&["--printf=.\\012a\\377b", "."])
+        .succeeds()
+        .stdout_is_bytes(expected_stdout);
+}
+
+#[test]
+fn test_printf_hex_3() {
+    let ts = TestScenario::new(util_name!());
+    ts.ucmd()
+        .args(&["--printf=\\x", "."])
+        .run()
```

Comment:
`run` doesn't tell me whether the command is expected to fail or succeed. And as it is expected to succeed, I would use `succeeds()`:
```suggestion
        .succeeds()
```